drivers/console/semihost_console.c

```diff
@@ -13,16 +13,23 @@ extern void __stdout_hook_install(int (*fn)(int));
 
 static int semihost_console_out(int ch)
 {
-	static unsigned char c;
-
-	c = ch;
-	__asm__ __volatile__ (
-		"movs	r1, %0\n"
-		"movs	r0, #3\n"
-		"bkpt	0xab\n"
-		:
-		: "r" (&c)
-		: "r0", "r1");
+	static unsigned short n;
+	static unsigned char buf[CONFIG_SEMIHOST_CONSOLE_OUTPUT_BUFFER_SIZE];
+
+	buf[n++] = ch;
+	buf[n] = 0;
+
+	if (ch == 0 || n >= sizeof(buf) - 2 || ch == '\r' || ch == '\n') {
```

Comment:
I liked @atx 's solution with a timer better, I don't want to think about whether I've put a CR / LF or a null-terminated string in the output, I don't think the other console drivers require that?
I also may want to print multiple lines in a single message, there is no reason it should `bkpt` after each line.